File: Grind 75 in Java/0232-implement-queue-using-stacks.java
class MyQueue {
    LinkedList<Integer> in = new LinkedList<Integer>();
    LinkedList<Integer> out = new LinkedList<Integer>();
    
    public void push(int x) {
        in.addLast(x);
    }
    
    public int pop() {
        refill();
        return out.removeLast();
    }
    
    public int peek() {
        refill();
        return out.peekLast();
    }

    private void refill() {
        if (out.isEmpty()) {
            while (!in.isEmpty())
                out.addLast(in.removeLast());
        }
    }
    
    public boolean empty() {
        return in.isEmpty() && out.isEmpty();
    }
}

File: leetcode in kotlin/0232-implement-queue-using-stacks.kt
class MyQueue() {
    
    val input = LinkedList<Int>()
    val output = LinkedList<Int>()

    fun push(x: Int) {
        input.addLast(x)
    }

    fun pop(): Int {
        if (output.isEmpty() ){
            refill()
        } 
        return output.removeLast()
    }

    fun peek(): Int {
        if(output.isEmpty()){
            refill()
        } 
        return output.peekLast()
    }

    private fun refill() {
        while (input.isNotEmpty())
            output.addLast(input.removeLast())
    }

    fun empty() = input.isEmpty() && output.isEmpty()

}
